x/poolmanager/taker_fee.go

```diff
@@ -13,9 +15,17 @@
 )
 
 func (k Keeper) GetDefaultTakerFee(ctx sdk.Context) sdk.Dec {
-	var defaultTakerFee sdk.Dec
-	k.paramSpace.Get(ctx, types.KeyDefaultTakerFee, &defaultTakerFee)
-	return defaultTakerFee
+	defaultTakerFeeBz := k.paramSpace.GetRaw(ctx, types.KeyDefaultTakerFee)
+	if !bytes.Equal(defaultTakerFeeBz, k.defaultTakerFeeBz) {
+		var defaultTakerFeeValue sdk.Dec
+		err := json.Unmarshal(defaultTakerFeeBz, &defaultTakerFeeValue)
+		if err != nil {
+			defaultTakerFeeValue = sdk.ZeroDec()
+		}
+		k.defaultTakerFeeBz = defaultTakerFeeBz
```

Comment:
Ah no, it was indeed ineffective. But that was because the keeper receiver was a struct not a pointer. (So the suggestion was wrong anyway)

Making the receiver a pointer fixed it